applications/tests/test_applications_new_schema_ws_queries.py

```diff
@@ -0,0 +1,394 @@
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "edges": [
+                {
+                    "node": {
+                        "id": to_global_id(
+                            WinterStorageApplicationNode._meta.name,
+                            winter_storage_application.id,
+                        ),
+                        "customer": None,
+                    }
+                }
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_no_customer_filter_false(
+    winter_storage_application, api_client
+):
+    winter_storage_application.customer = None
+    winter_storage_application.save()
+
+    query = WS_APPLICATIONS_WITH_NO_CUSTOMER_FILTER_QUERY % "false"
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {"winterStorageApplications": {"edges": []}}
+
+
+WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY = """
+query APPLICATIONS {
+    winterStorageApplications(statuses: [%s]) {
+        edges {
+            node {
+                id
+                status
+            }
+        }
+    }
+}
+"""
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter(
+    handled_ws_application, api_client
+):
+
+    status_enum_str = ApplicationStatus.HANDLED.name
+
+    query = WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY % status_enum_str
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {
+        "winterStorageApplications": {
+            "edges": [
+                {
+                    "node": {
+                        "id": to_global_id(
+                            WinterStorageApplicationNode._meta.name,
+                            handled_ws_application.id,
+                        ),
+                        "status": status_enum_str,
+                    }
+                }
+            ]
+        }
+    }
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter_empty(
+    handled_ws_application, api_client
+):
+    status_enum_str = ApplicationStatus.PENDING.name
+
+    query = WINTER_STORAGE_APPLICATIONS_WITH_STATUSES_FILTER_QUERY % status_enum_str
+
+    executed = api_client.execute(query)
+
+    assert executed["data"] == {"winterStorageApplications": {"edges": []}}
+
+
+@pytest.mark.parametrize(
+    "api_client",
+    ["berth_services", "berth_handler", "berth_supervisor"],
+    indirect=True,
+)
+def test_winter_storage_applications_statuses_filter_invalid_enum(
+    handled_ws_application, api_client
+):
```

Comment:
For cases like this when you're not necessarily testing that the user has the right permissions, you can use the `superuser_api_client`, you save a couple of unnecessary tests and emphasize that the "fail" you're checking for doesn't directly concern user permissions.

Reply:
good point, I change it.